File: JavaFundamentals/03_Arrays/ArraysExercise/src/_04ArrayRotation.java
import java.util.Arrays;
import java.util.Scanner;

public class _04ArrayRotation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int[] numbers = Arrays
                .stream(scanner.nextLine().split(" "))
                .mapToInt(e -> Integer.parseInt(e))
                .toArray();

        int rotations = Integer.parseInt(scanner.nextLine());

        for (int i = 0; i < rotations; i++) {
            int currentNum = numbers[0];

            for (int j = 0; j < numbers.length - 1; j++) {
                numbers[j] = numbers[j + 1];
            }

            numbers[numbers.length - 1] = currentNum;
        }

        for (int number : numbers) {
            System.out.print(number + " ");
        }
    }
}

File: JavaScriptFundamentals/03_Arrays/03_2_ArraysExercises/04_ArrayRotation.js
function arrayRotation(numbers, n) {
    for (let i = 0; i < n; i++) {
        let initialFirstElement = numbers[0];
        
        for (let i = 0; i < numbers.length - 1; i++) {
            numbers[i] = numbers[i + 1];
        }
        numbers[numbers.length - 1] = initialFirstElement;
    }

    console.log(numbers.join(' '));
}

arrayRotation(
    [51, 47, 32, 61, 21], 2
);